### ISSUE TITLE
e2e local test is broken

### ISSUE THREAD

**Issue by crazyxy** [istio/istio#16001]
Cannot run istio e2e test locally. It seems to be related with these four commit:

aa330ced4f3f1b65cc661b88b405ff27364764b0
fb643b6d4ffebff2f36bda62b6becaf7acf8980c
32e66a7be34519d3fe23fd9fbd8c6739cfd6ec76
be7e0a3a8dc479fd6dfb7421ba02437655c946fb

I reverted these four commits locally and rerun e2e_simple, it passed.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

**Steps to reproduce the bug**

```bash
git pull origin master
make clean && make docker && make push && make e2e_simple
```

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by ericvn** [comment]
I was seeing a similar problem in the integration framework running locally. I got things to work (with at least one failing test) by clearing all the local docker base images and building and pushing again. I suspect a base image was updated and docker isn't pulling that base image and that is causing the failure.

**Comment by clyang82** [comment]
Yes. You can see that the commits to enable the istio components can be able to run as nonroot. so you have to clean all local docker images and build again since the image are updated. Thanks.

Let me know if you still have `local e2e test is broken` issue. Thanks.

**Comment by ericvn** [comment]
A fix for this issue is in https://github.com/istio/istio/pull/14623.

**Comment by ericvn** [comment]
Not sure if this is something where we want to make sure we always pull the base images for particular images or not. Would a `--pull` in the DOCKER_RULE (in Istio-docker.mk) cause too much additional overhead? Maybe this only occurs at random times and a comment on discuss would be enough?

**Comment by ericvn** [comment]
Thinking about this more, the `--pull` would get the base image pulled for each image which would be excessive. It would be better to make sure it's done once for each base image before any of the images are built.

**Comment by rshriram** [comment]
I am getting the same error. This is not  just for local e2e. I cannot run istio on minikube (none driver, on linux) as it has same issue. I even disabled selinux but nothing seems to have changed. If these changes required updates to our install yamls, then they shouyld have been made with the PRs above. Until then these changes need to be reverted.

**Comment by rshriram** [comment]
fwiw, I cannot even run the docker container on OSX directly. I get the same error on both kube and osx:
```
standard_init_linux.go:211: exec user process caused "permission denied"
```

docker run -u 1001:1001 didn't work either.. Also tried the K8S securityContext.runAsUser 1001, runAsGroup: 1001.. No change.. 

The only time I have been able to run on both minikube and OSX is when I set the runAsUser to 0:0, which is not a nice solution for people running Istio locally for dev/test/even trial.

**Comment by rshriram** [comment]
I have reverted pilot/galley.. need to fix mixer https://github.com/istio/istio/pull/15887 .. cc @mandarjog

**Comment by rshriram** [comment]
reverted mixer as well..

**Comment by clyang82** [comment]
@rshriram sorry for inconvenience. this PR is to resolve this issue - https://github.com/istio/istio/pull/16014 as I explained in this PR, I am not sure that why it is working in some versioned container runtime. Anyway, that should be an issue with wrong permission.

**Comment by mandarjog** [comment]
Thanks Shriram. We need to have a minikube test, though why not just support kube-kind?

**Comment by sdake** [comment]
From reading the PR, it appears that docker 18.03 produces different results than docker 18.09.

This requires more investigation - marking as a P0 release blocker.

**Comment by clyang82** [comment]
I think that is related with the binary permission. I did test before, even the binary does not have execute permission, it can be executed in 18.03 runtime. We should give the binary correct permission to ensure it can be executed in each docker runtime.

**Comment by howardjohn** [comment]
What is the status of this? It is marked as a release blocker. All changes are reverted right? Can this be closed/dropped priority

**Comment by howardjohn** [comment]
Since this has had no attention for 15 days, I am going to assume this is not really a release blocker and has been fixed by reverting

**Comment by clyang82** [comment]
Yes. That is not a blocker for this release. And I think we can close this issue since all changes are reverted. I am working on new PR to cover this case. Thanks.
